cli/src/commands/bookmark/mod.rs

```diff
@@ -181,15 +182,19 @@ fn find_trackable_remote_bookmarks<'a>(
     }
 }
 
-fn is_fast_forward(repo: &dyn Repo, old_target: &RefTarget, new_target_id: &CommitId) -> bool {
+fn is_fast_forward(
+    repo: &dyn Repo,
+    old_target: &RefTarget,
+    new_target_id: &CommitId,
+) -> Result<bool, CommandError> {
     if old_target.is_present() {
         // Strictly speaking, "all" old targets should be ancestors, but we allow
         // conflict resolution by setting bookmark to "any" of the old target
         // descendants.
-        old_target
-            .added_ids()
-            .any(|old| repo.index().is_ancestor(old, new_target_id))
+
+        let is_ff = any_is_ancestor(repo.index(), old_target.added_ids(), new_target_id)?;
```

Comment:
nit: I don't think "any_is_ancestor" is common operation. For example, if we were strict, here we should check if "all" targets are ancestors.

Maybe we can instead add a few generic helper functions (e.g. `fallible_any(iter, predicate) -> Result<bool>)`? We can also map items to `Result` to use `Itertools::process_results()`. It wouldn't be as readable as the infallible form, though.

```rust
iter
    .map(|item| fallible_fn(item)) // T -> Result<U, E> (e.g. U = bool)
    .process_results(|predicates| predicates.any(|p| p))
```
https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.process_results

Reply:
I see. Also, good to know about `process_results` -- I'll play with that, and compare it with the alternative of some new generic helper functions. Thanks!